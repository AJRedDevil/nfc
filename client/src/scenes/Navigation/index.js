// npm packages
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

// our packages
import LinkTab from '../../components/LinkTab';
import {setRoute, setDefaultRoute} from '../../services/navigation/actions';
import {Logger} from '../../utils';

const NAVIGATION_LINKS = [
  {path: '', text: 'Dashboard'},
  {path: 'classicleague', text: 'Classic League'},
  {path: 'h2hleague', text: 'H2H League'},
  {path: 'gwtable', text: 'GW Overall'},
];

class Navigation extends Component {
  componentWillMount() {
    Logger.info(`setDefaultTab: ${this.props.pathname}`, 'Navigation.cwm');
    this.props.setDefaultTab(this.props.pathname);
  }

  handleTabClick = e => {
    const nextTab = e.target.getAttribute('identifier');
    this.props.setCurrentTab(nextTab);
  };

  render() {
    return (
      <aside className="column is-2 is-narrow-mobile is-fullheight section is-hidden-mobile">
        <p className="menu-label is-hidden-touch">NFC</p>
        <ul className="menu-list">
          {NAVIGATION_LINKS.map(link => (
            <LinkTab
              key={link.text}
              link={link}
              currentTab={this.props.navigation.currentTab}
              onTabClick={this.handleTabClick}
            />
          ))}
        </ul>
      </aside>
    );
  }
}
Navigation.propTypes = {
  navigation: PropTypes.shape({
    currentTab: PropTypes.string,
  }),
  setCurrentTab: PropTypes.func.isRequired,
  setDefaultTab: PropTypes.func.isRequired,
  pathname: PropTypes.string,
};

const mapStateToProps = state => ({
  navigation: state.Navigation,
  pathname: state.router.location.pathname,
});

const mapDispatchToProps = dispatch => ({
  setCurrentTab: nextRoute => dispatch(setRoute(nextRoute)),
  setDefaultTab: pathname => dispatch(setDefaultRoute(pathname)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Navigation);
