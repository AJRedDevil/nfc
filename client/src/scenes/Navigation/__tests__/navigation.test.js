// npm packages
import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import configureMockStore from 'redux-mock-store';

// our packages
import Navigation from '../index';

// create mock store
const mockStore = configureMockStore();

test('# Navigation', () => {
  const store = mockStore({
    Navigation: {currentTab: 'test'},
    router: {
      location: {
        pathname: 'http://localhost:8080',
      },
    },
  });
  const component = renderer.create(
    <MemoryRouter>
      <Navigation store={store} />
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
