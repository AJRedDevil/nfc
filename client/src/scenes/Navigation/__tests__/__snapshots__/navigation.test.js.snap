// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Navigation 1`] = `
<aside
  className="column is-2 is-narrow-mobile is-fullheight section is-hidden-mobile"
>
  <p
    className="menu-label is-hidden-touch"
  >
    NFC
  </p>
  <ul
    className="menu-list"
  >
    <li
      className=""
    >
      <a
        aria-current="page"
        className=""
        href="/"
        identifier=""
        onClick={[Function]}
        style={Object {}}
      >
        Dashboard
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current={null}
        className={undefined}
        href="/classicleague"
        identifier="classicleague"
        onClick={[Function]}
        style={undefined}
      >
        Classic League
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current={null}
        className={undefined}
        href="/h2hleague"
        identifier="h2hleague"
        onClick={[Function]}
        style={undefined}
      >
        H2H League
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current={null}
        className={undefined}
        href="/gwtable"
        identifier="gwtable"
        onClick={[Function]}
        style={undefined}
      >
        GW Overall
      </a>
    </li>
  </ul>
</aside>
`;
